code/datums/mutations/body.dm

```diff
@@ -375,3 +375,64 @@
 			owner.SetStun(owner.AmountStun()*2)
 			owner.visible_message("<span class='danger'>[owner] tries to stand up, but trips!</span>", "<span class='userdanger'>You trip over your own feet!</span>")
 			stun_cooldown = world.time + 300
+
+/datum/mutation/human/strongwings
+	name = "Strengthened Wings"
+	desc = "Subject's wing muscle volume rapidly increases."
+	quality = POSITIVE
+	difficulty = 12
+	instability = 15
+	species_allowed = list(SPECIES_APID, SPECIES_MOTH)
+
+/datum/mutation/human/strongwings/on_acquiring()
+	if(..())
+		return
+	if(locate(/obj/item/organ/wings/bee) in owner.internal_organs)
+		var/obj/item/organ/wings/bee/wings = locate(/obj/item/organ/wings/bee) in owner.internal_organs
+		wings.jumpdist = 6 * GET_MUTATION_POWER(src)
+		to_chat(owner, "<span class='notice'>Your wings feel stronger.</span>")
+	else if(locate(/obj/item/organ/wings/moth) in owner.internal_organs)
+		var/obj/item/organ/wings/moth/wings = locate(/obj/item/organ/wings/moth) in owner.internal_organs
+		wings.flight_level = WINGS_FLYING
+		to_chat(owner, "<span class='notice'>Your wings feel stronger.</span>")
+		wings.Refresh(owner)
+	else
+		to_chat(owner, "<span class='warning'>You don't have wings to strengthen!")
+
+/datum/mutation/human/strongwings/on_losing()
+	if(..())
+		return
+	if(locate(/obj/item/organ/wings/bee) in owner.internal_organs)
+		var/obj/item/organ/wings/bee/wings = locate(/obj/item/organ/wings/bee) in owner.internal_organs
+		wings.jumpdist = 3
+		to_chat(owner, "<span class='warning'>Your wings feel weak.</span>")
+	else if(locate(/obj/item/organ/wings/moth) in owner.internal_organs)
+		var/obj/item/organ/wings/moth/wings = locate(/obj/item/organ/wings/moth) in owner.internal_organs
+		wings.flight_level = WINGS_FLIGHTLESS
```

Comment:
should use initial(). also, notably, this conflicts with other flight boosts... would be nice if you fixed that